plugins/pulp_rpm/plugins/importers/yum/associate.py

```diff
@@ -196,12 +210,27 @@ def copy_rpms(units, source_repo, dest_repo, import_conduit, copy_deps, solver=N
     """
     unit_set = set()
 
+    failed_signature_check = 0
     for unit in units:
         # we are passing in units that may have flattened "provides" metadata.
         # This flattened field is not used by associate_single_unit().
-        repo_controller.associate_single_unit(dest_repo, unit)
+        if rpm_parse.signature_enabled(config):
+            if unit.downloaded:
+                try:
+                    rpm_parse.verify_signature(unit, config)
+                except PulpCodedException, e:
```

Comment:
To prepare for python 3 and make that transition slightly easier, it's good to get in the habit of using the ``except Exception as e`` syntax.